### PR TITLE
Worked on Notification and Worker 

### PR DESCRIPTION
**Pull request by Arunshaik2001**
created a worker for showing notification
created notification helper to build notification
added two buttons for 1 hour to postpone by an hour and one is done.

I used @midasjr7 time picker UI for setting the time for testing purpose.

### CODE REVIEW

**Comment by sumanabhi** [comment]
> created a worker for showing notification created notification helper to build notification added two buttons for 1 hour to postpone by an hour and one is done.
> 
> I used @midasjr7 time picker UI for setting the time for testing purpose.

First of all, why the .idea and .misc files you're pushing? 

Please check and remove it from PR.

try checking why it is not coming inside the .gitignore in your system.

File: app/src/main/java/com/hadiyarajesh/notex/MyApplication.kt [hunk 1]
```diff
@@ -1,7 +1,21 @@
...
 
 import android.app.Application
+import androidx.hilt.work.HiltWorkerFactory
+import androidx.work.Configuration
 import dagger.hilt.android.HiltAndroidApp
+import javax.inject.Inject
 
 @HiltAndroidApp
-class MyApplication: Application()
+class MyApplication: Application(),Configuration.Provider{
```
Comment (sumanabhi) [inline app/src/main/java/com/hadiyarajesh/notex/MyApplication.kt:10]: nit: try doing a formatting before pushing.

File: app/src/main/java/com/hadiyarajesh/notex/database/entity/Reminder.kt [hunk 2]
```diff
@@ -10,7 +10,7 @@ import java.time.Instant
 @JsonClass(generateAdapter = true)
 data class Reminder(
     @PrimaryKey(autoGenerate = true)
-    val reminderId: Long? = null,
+    val reminderId: Long? =null,
```
Comment (sumanabhi) [inline app/src/main/java/com/hadiyarajesh/notex/database/entity/Reminder.kt:13]: why this change? the previous formatting is good.

check your format setting in your IDE.

File: app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -28,6 +28,14 @@
                 <category android:name="android.intent.category.LAUNCHER" />
             </intent-filter>
         </activity>
+
+        <provider
+            android:name="androidx.startup.InitializationProvider"
+            android:authorities="${applicationId}.androidx-startup"
+            tools:node="remove">
```
Comment (sumanabhi) [inline app/src/main/AndroidManifest.xml:35]: why is this required?

File: app/src/main/java/com/hadiyarajesh/notex/reminder/notification/NotificationBroadCastReceiver.kt [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+@AndroidEntryPoint
+class NotificationBroadCastReceiver : HiltBroadcastReceiver() {
+
+    @Inject
+    lateinit var reminderWorkManager: ReminderWorkManager
+
+
+    override fun onReceive(context: Context, intent: Intent) {
+        super.onReceive(context, intent)
+        if (intent.action == context.resources.getString(R.string.done_action)) {
```
Comment (sumanabhi) [inline app/src/main/java/com/hadiyarajesh/notex/reminder/notification/NotificationBroadCastReceiver.kt:24]: better use case with when().

File: app/src/main/java/com/hadiyarajesh/notex/reminder/notification/NotificationHelper.kt [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+import android.content.Context
+import android.content.Intent
+import androidx.core.app.NotificationCompat
+import androidx.core.app.NotificationManagerCompat
+import com.hadiyarajesh.notex.MainActivity
+import com.hadiyarajesh.notex.R
+import javax.inject.Inject
+import kotlin.random.Random
+
+class NotificationHelper @Inject constructor() {
```
Comment (sumanabhi) [inline app/src/main/java/com/hadiyarajesh/notex/reminder/notification/NotificationHelper.kt:14]: what are we injecting here?

Constructor() is fully empty

**Review by sumanabhi — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 6]
```diff
@@ -16,6 +16,7 @@ local.properties
 
 # Android Studio / IntelliJ IDEA
 *.iws
+/.idea
```
Comment (sumanabhi) [inline .gitignore:19]: your files are still under here.

So, I guess the changes you have made are not working correctly. and also the changes are already there to ignore these folders. Please check once again before pushing.

**Review by sumanabhi — CHANGES_REQUESTED** [review]

File: app/src/main/AndroidManifest.xml (hunk 3 above)
Comment (Arunshaik2001) [reply app/src/main/AndroidManifest.xml:35]: Because of this error when I used hiltWorker
```
java.lang.NoSuchMethodException: com.hadiyarajesh.notex.reminder.worker.ReminderWorker.<init> [class android.content.Context, class androidx.work.WorkerParameters]
        at java.lang.Class.getConstructor0(Class.java:2332)
        at java.lang.Class.getDeclaredConstructor(Class.java:2170)
        at androidx.work.WorkerFactory.createWorkerWithDefaultFallback(WorkerFactory.java:95)
        at androidx.work.impl.WorkerWrapper.runWorker(WorkerWrapper.java:245)
        at androidx.work.impl.WorkerWrapper.run(WorkerWrapper.java:137)
        at androidx.work.impl.utils.SerialExecutor$Task.run(SerialExecutor.java:91)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:923)
E/WM-WorkerWrapper: Could not create Worker com.hadiyarajesh.notex.reminder.worker.ReminderWorker
```

File: app/src/main/java/com/hadiyarajesh/notex/reminder/notification/NotificationBroadCastReceiver.kt (hunk 4 above)
Comment (Arunshaik2001) [reply app/src/main/java/com/hadiyarajesh/notex/reminder/notification/NotificationBroadCastReceiver.kt:24]: ok

File: app/src/main/java/com/hadiyarajesh/notex/reminder/notification/NotificationHelper.kt (hunk 5 above)
Comment (Arunshaik2001) [reply app/src/main/java/com/hadiyarajesh/notex/reminder/notification/NotificationHelper.kt:14]: because it is injected in worker

File: .gitignore (hunk 6 above)
Comment (Arunshaik2001) [reply .gitignore:19]:
```
public abstract static class SingletonC implements MyApplication_GeneratedInjector,
                         ^
      com.hadiyarajesh.notex.reminder.notification.NotificationHelper is injected at
```
